camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/dsl/methods/BrooklynDslCommon.java

```diff
@@ -231,6 +233,16 @@ public DslFormatString(String pattern, Object ...args) {
         }
 
         @Override
+        public int hashCode() {
+            return Objects.hashCode(pattern, args);
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            return EqualsBuilder.reflectionEquals(this, obj);
```

Comment:
Personally I feel `reflectionEquals` is fragile - if someone adds another field it's going to be automatically incorporated into `equals` but not into `hashCode` and we've broken the contract. Is there any reason to prefer the shortcut over just defining `equals` normally?